coderd/database/dbauthz/dbauthz.go

```diff
@@ -1106,6 +1106,15 @@ func (q *querier) BulkMarkNotificationMessagesSent(ctx context.Context, arg data
 	return q.db.BulkMarkNotificationMessagesSent(ctx, arg)
 }
 
+func (q *querier) ClaimPrebuild(ctx context.Context, newOwnerID database.ClaimPrebuildParams) (database.ClaimPrebuildRow, error) {
+	if err := q.authorizeContext(ctx, policy.ActionUpdate, rbac.ResourceWorkspace); err != nil {
```

Comment:
Self Review: Consider explicitly checking for the prebuilds user or a prebuild specific role in the context.

Reply:
I'm working on the authz improvements, but lets not delay additional review until this is done. cc @dannykopping

Reply:
@Emyrk  Fixed, now it checks:
```go
Asserts(
	rbac.ResourceWorkspace.WithOwner(user.ID.String()).InOrg(org.ID), policy.ActionCreate,
	template, policy.ActionRead,
	template, policy.ActionUse,
)
```